### PR TITLE
BUG: update export code

### PR DESCRIPTION
**Pull request by fedorov**
Re #99 this accounts for the API changes in the Slicer core

WIP: do not merge!

### CODE REVIEW

**Comment by fedorov** [comment]
when I run export using this branch, I get this popup:

image

@che85 I am not familiar with this type of error reporting - any idea how to figure out what line causes it?

**Comment by che85** [comment]
I use PyCharm for debugging ....

**Comment by fedorov** [comment]
have you seen anything like this before (python error reported in a popup)?

**Comment by che85** [comment]
Yes that's cause, because I am using try/except code snippet here [1]

[1] https://github.com/QIICR/Reporting/blob/master/Py/Reporting.py#L347-L354

**Comment by che85** [comment]
Just remove that try/except and try again....

**Comment by fedorov** [comment]
ah, I see - thanks!

File: Py/Reporting.py [hunk 1]
```diff
@@ -769,71 +769,72 @@ def filterEmptySegments(self):
...
-    region = slicer.vtkSlicerTerminologyType()
-    self.terminologyLogic.GetRegionInAnatomicContext(anatomicContextName, regionName, region)
-    segmentData["AnatomicRegionSequence"] = self.getJSONFromVtkSlicerCodeSequence(region)
-
-    modifierName = self.getTagValue(segment, segment.GetAnatomicRegionModifierTagName())
-    if modifierName != "":
-      modifier = slicer.vtkSlicerTerminologyType()
-      self.terminologyLogic.GetRegionModifierInAnatomicRegion(anatomicContextName, regionName, modifierName, modifier)
-      segmentData["AnatomicRegionModifierSequence"] = self.getJSONFromVtkSlicerCodeSequence(modifier)
+    regionModifierObject = self.terminologyEntry.GetAnatomicRegionModifierObject()
```
Comment (che85) [inline Py/Reporting.py]: self.terminologyEntry has never been initialized.

File: Py/Reporting.py [hunk 2]
```diff
@@ -769,71 +769,72 @@ def filterEmptySegments(self):
...
       if self.statistics[segmentID, "GS voxel count"] == 0:
         continue
       segmentData = dict()
       segmentData["LabelID"] = 1
       segment = self.segmentationNode.GetSegmentation().GetSegment(segmentID)
-      category = self.getTagValue(segment, segment.GetTerminologyCategoryTagName())
-      segmentData["SegmentDescription"] = category if category != "" else self.statistics[segmentID, "Segment"]
+
+      terminologyWidget = slicer.qSlicerTerminologyNavigatorWidget()
+      terminologyEntry = slicer.vtkSlicerTerminologyEntry()
```
Comment (che85) [inline Py/Reporting.py]: self.terminologyEntry should be initialized here i guess..

**Comment by fedorov** [comment]
Still not ready to merge - I am looking into SR export issues now ...

**Comment by fedorov** [comment]
Now there is another error, but it seems like it is unrelated to this issue

image

**Comment by che85** [comment]
What's the modality value for the image?

**Comment by fedorov** [comment]
CT

**Comment by che85** [comment]
That's strange

**Comment by che85** [comment]
Maybe there is a blank somewhere?

**Comment by che85** [comment]
Otherwise I can just do that tomorrow once I have the new nightly build...

**Comment by fedorov** [comment]
I was using the CT series from this dataset for testing: https://fedorov.gitbooks.io/rsna2016-qirr-dicom4qi/content/instructions/seg.html#test-dataset-3

**Comment by fedorov** [comment]
Let's merge and investigate separately?

**Comment by che85** [comment]
Ok I am going to work on this locally on my computer

**Pull request merged**